app/assets/javascripts/discourse/tests/integration/components/d-select-test.gjs

```diff
@@ -0,0 +1,70 @@
+import { render } from "@ember/test-helpers";
+import { module, test } from "qunit";
+import DSelect, { NO_VALUE_OPTION } from "discourse/components/d-select";
+import { setupRenderingTest } from "discourse/tests/helpers/component-test";
+import dselectHelper from "discourse/tests/helpers/d-select-helper";
+import { i18n } from "discourse-i18n";
+
+module("Integration | Component | d-select", function (hooks) {
+  setupRenderingTest(hooks);
+
+  test("@onChange", async function (assert) {
+    const handleChange = (value) => {
+      assert.step(value);
+    };
+
+    await render(<template>
+      <DSelect @onChange={{handleChange}} as |select|><select.Option
+          @value="foo"
+        >foo</select.Option></DSelect>
```

Comment:
the templates here could use a bit of reformatting

```suggestion
      <DSelect @onChange={{handleChange}} as |select|>
        <select.Option @value="foo">foo</select.Option>
      </DSelect>
```

(and this example could possibly be better if the value wasn't the same as the label)